### ISSUE TITLE
Keyword Tag Duplicate Frames within Project

### ISSUE THREAD

**Issue by latenitefilms** [fcpxhacks/fcpxhacks#86]
Ideally this would use plist XML, however it could also be done with FCPXML.

Assuming we're using plist XML:

- When you press a shortcut key it will detect whether your focus is on the timeline or browser. If the browser it will select everything in the timeline and copy it to the pasteboard. If it's the browser, it will open the selected project in the timeline, then select all and again, copy it to the pasteboard.
- Then the tricky stuff... we need to read the plist data from the pasteboard, then "process" it to add the appropriate keywords (assuming this is technically possible).
- Once "processed" we then place this data back on the pasteboard, select everything in the Final Cut Pro timeline, delete, then paste.

Have discussed this feature a few times with Alex Gollner on Facebook.

**Comment by randomeizer** [comment]
I don't think you can add keywords to clips in the timeline, only in the
Browser. As such, you would have to match-frame each duplicate in the
browser, add the keyword, flip back to the timeline.

**Comment by latenitefilms** [comment]
Correct - but it MIGHT be possible if you're manipulating the clipboard XML. Otherwise, I think it's definitely possible using FCPXML (as this is basically exactly what Sync-N-Link, etc. do).

**Comment by randomeizer** [comment]
Not on the Timeline. Sync--link works with Libraries.

As far as I know, keywords only exist at the Event level.

**Comment by latenitefilms** [comment]
You're absolutely right. I just assumed that FCPX was smart enough to keep track of keywords when you export projects, but apparently not. I just exported a project out of one library and brought it into another, and none of the keywords have come across. Doh!

I guess the only way to make this work then would be to drag an Event to the drop zone, and prompt the user which project they want to apply the tags to (or maybe it just applies to all projects within the event)? I just tested and exporting an event as FCPXML definitely retains all the keywords.
